frontend/src/hooks/useShownButtons.tsx

```diff
@@ -0,0 +1,36 @@
+import { useCallback, useEffect, useState } from 'react';
+
+/**
+ * React hook to determine the number of buttons to be shown on the toolbar.
+ * @returns {number} - The number of buttons to be shown on the toolbar.
+ */
+const useShownButtons = (): number => {
+  const getNumberOfButtons = useCallback((width: number) => {
+    // Minimum width: 354 = 107 audio selector + 107 video selector + (48 + 12) overflow menu + 48 exit button + 32px padding
+    const widthForEssentialButtons = 107 + 107 + (48 + 12) + 48 + 32;
```

Comment:
We can go with this for now, it's much cleaner than before, but ideally we should a more generic way to determine available space